src/main/kotlin/com/ing/dlt/zkkrypto/ecc/schnorr/SchnorrSignature.kt

```diff
@@ -2,35 +2,33 @@ package com.ing.dlt.zkkrypto.ecc.schnorr
 
 import com.ing.dlt.zkkrypto.ecc.EllipticCurve
 import com.ing.dlt.zkkrypto.ecc.EllipticCurvePoint
+import com.ing.dlt.zkkrypto.ecc.HashEnum
 import com.ing.dlt.zkkrypto.ecc.curves.AltBabyJubjub
 import com.ing.dlt.zkkrypto.ecc.schnorr.FixedGenerators.altBabyJubjubFixedGenerators
 import com.ing.dlt.zkkrypto.util.asUnsigned
 import org.bouncycastle.crypto.digests.Blake2bDigest
 import org.bouncycastle.crypto.digests.Blake2sDigest
+import java.lang.IllegalArgumentException
 import java.math.BigInteger
 import java.security.SecureRandom
 import kotlin.random.Random.Default.nextBytes
 
-enum class Hash{
-    BLAKE2S,
-    BLAKE2B
-}
-
 /**
  * Implementation of Schnorr signatures
  * This implementation uses AltBabyJubJub curve and Blake2 hash
  */
 
-class SchnorrSignature (
+class SchnorrSignature(
```

Comment:
Have you already looked at how to integrate this into Corda through/as a Security Provider? I thought to be able to do that, the `Signature` implementation has to be a `SignatureSpi`: 
```
 * This class defines the <i>Service Provider Interface</i> (<b>SPI</b>)
 * for the {@code Signature} class, which is used to provide the
 * functionality of a digital signature algorithm. Digital signatures are used
 * for authentication and integrity assurance of digital data.
 *.
 * <p> All the abstract methods in this class must be implemented by each
 * cryptographic service provider who wishes to supply the implementation
 * of a particular signature algorithm.
```
And this would also require you to make the keys as standard keys: `java.security.PublicKey` and `java.security.PrivateKey`.

That said, I am not a specialist in this area, I just looked at where Corda gets its supported signature schemes from and how they are used. Perhaps you have found another, easier way in which the current API you designed can be used.